packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -0,0 +1,358 @@
+   */
+  inputAttributes: PropTypes.object,
+  /**
+   * An ISO 8601 string representation of the maximum date time.
+   */
+  maxDateTime: PropTypes.string,
+  /**
+   * An ISO 8601 string representation of the minimum date time.
+   */
+  minDateTime: PropTypes.string,
+  /**
+   * Name of the date input. The name should be unique.
+   */
+  name: PropTypes.string.isRequired,
+  /**
+   * A callback function to execute when a valid date is selected or entered.
+   * The first parameter is the event. The second parameter is the changed input value.
+   */
+  onChange: PropTypes.func,
+    /**
+   * A callback function to execute when a change is made in the date or time input.
+   * The first parameter is the event. The second parameter is the changed input value.
+   */
+  onChangeRaw: PropTypes.func,
+  /**
+   * A callback function to let the containing component (e.g. modal) to regain focus.
+   */
+  releaseFocus: PropTypes.func,
+  /**
+   * A callback function to request focus from the containing component (e.g. modal).
+   */
+  requestFocus: PropTypes.func,
+  /**
+   * An ISO 8601 string representation of the initial value to show in the date and time inputs.
+   */
+  value: PropTypes.string,
+};
+
+const defaultProps = {
+  excludeDates: undefined,
+  filterDate: undefined,
+  includeDates: undefined,
+  inputAttributes: undefined,
+  maxDateTime: undefined,
+  minDateTime: undefined,
+  name: undefined,
+  onChange: undefined,
+  onChangeRaw: undefined,
+  releaseFocus: undefined,
+  requestFocus: undefined,
+  value: undefined,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+const keyCodes = {
+  ARROWDOWN: 40,
+};
+
+class DateTimePicker extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dateTime: DateTimeUtil.createSafeDate(props.value),
+      isAmbiguousTime: false,
+      isTimeClarificationOpen: false,
+    };
+
+    // Track the dateValue and timeValue outside of the react state to limit the number of renderings that occur.
+    this.dateValue = '';
+    this.timeValue = '';
+
+    this.handleDateChange = this.handleDateChange.bind(this);
+    this.handleDateChangeRaw = this.handleDateChangeRaw.bind(this);
+    this.handleTimeChange = this.handleTimeChange.bind(this);
+    this.handleOnSelect = this.handleOnSelect.bind(this);
+    this.handleOnDateBlur = this.handleOnDateBlur.bind(this);
+    this.handleOnTimeBlur = this.handleOnTimeBlur.bind(this);
+
+    this.handleDaylightSavingButtonClick = this.handleDaylightSavingButtonClick.bind(this);
+    this.handleStandardTimeButtonClick = this.handleStandardTimeButtonClick.bind(this);
+  }
+
+  componentWillMount() {
+    const dateTime = DateTimeUtil.createSafeDate(this.props.value);
```

Comment:
Is this value different than `this,state.dateTime` declared on line 103?

Reply:
They both should have the same value when mounting. Updated to use `this.state.dateTime`.  https://github.com/cerner/terra-core/pull/768/commits/1823173fe270eecfe65c432c23a279e77cdeacde

Also added documentation to clarify the difference between `this.state.dateTime` and `this.dateValue`/`this.timeValue`. https://github.com/cerner/terra-core/pull/768/commits/3763caa7628bc29b961a755d2d760b627a28bbfa